server/resource_server/controller/controller_test.go

```diff
@@ -1852,3 +1857,144 @@ func setMockServiceInContext(req *http.Request) *http.Request {
+
+	req, err := http.NewRequest("POST", "/api/v1/register-user-precheck", bytes.NewBuffer(requestBody))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	mockService := &MockService{
+		registerUserPrecheck: func(req dto.RegisterUserPrecheckDTO, iterCount int) (string, error) {
+			assert.Equal(t, "test_user", req.Username, "Username should match")
+			assert.Equal(t, 4096, iterCount, "Iteration count should match")
+
+			return userSalt, nil
+		},
+	}
+
+	req = req.WithContext(context.WithValue(req.Context(), "service", mockService))
+
+	t.Setenv("SCRAM_ITERATION_COUNT", "4096")
+
+	rr := httptest.NewRecorder()
+
+	Ctl.RegisterUserPrecheck(rr, req)
+
+	assert.Equal(t, http.StatusCreated, rr.Code, "Response code should be 201 Created")
+
+	response := decodeResponseBodyForResponse(t, rr)
+
+	assert.Equal(t, "User is successfully registered", response.Message, "Response message should match")
+}
+
+func TestRegisterUserPrecheck_InvalidMethod(t *testing.T) {
+	req, err := http.NewRequest("GET", "/api/v1/register-user-precheck", nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	rr := httptest.NewRecorder()
+
+	Ctl.RegisterUserPrecheck(rr, req)
+
+	assert.Equal(t, http.StatusMethodNotAllowed, rr.Code, "Expected HTTP 405 Method Not Allowed")
+}
+
+func TestRegisterUserPrecheck_InvalidIterationCount(t *testing.T) {
+	requestBody := []byte(`{
+		"username": "test_user"
+	}`)
+
+	req, err := http.NewRequest("POST", "/api/v1/register-user-precheck", bytes.NewBuffer(requestBody))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	mockService := &MockService{}
+	req = req.WithContext(context.WithValue(req.Context(), "service", mockService))
+
+	t.Setenv("SCRAM_ITERATION_COUNT", "invalid_value")
+
+	rr := httptest.NewRecorder()
+
+	Ctl.RegisterUserPrecheck(rr, req)
+
+	assert.Equal(t, http.StatusInternalServerError, rr.Code, "Expected HTTP 500 Internal Server Error")
+}
+
+func TestRegisterUserPrecheck_InvalidJSON(t *testing.T) {
+	requestBody := []byte(`invalid_json`)
+
+	req, err := http.NewRequest("POST", "/api/v1/register-user-precheck", bytes.NewBuffer(requestBody))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	mockService := &MockService{}
+	req = req.WithContext(context.WithValue(req.Context(), "service", mockService))
+
+	t.Setenv("SCRAM_ITERATION_COUNT", "4096")
+
+	rr := httptest.NewRecorder()
+
+	Ctl.RegisterUserPrecheck(rr, req)
+
+	assert.Equal(t, http.StatusBadRequest, rr.Code, "Expected HTTP 400 Bad Request")
+}
+
+func TestRegisterUserPrecheck_MissingRequiredFields(t *testing.T) {
+	reqBody := []byte(`{
+		"other_field": "some value"
+	}`)
+
+	mockService := &MockService{
+		registerUserPrecheck: func(request dto.RegisterUserPrecheckDTO, iterCount int) (string, error) {
+			return "", nil
+		},
+	}
+
+	req := httptest.NewRequest("POST", "/api/register-user-precheck", bytes.NewBuffer(reqBody))
+	req = req.WithContext(context.WithValue(req.Context(), "service", mockService))
+	req.Header.Set("Content-Type", "application/json")
+
```

Comment:
Here you should set `t.Setenv("SCRAM_ITERATION_COUNT", "4096")`, currently the test fails because of that (I think it tries to parse an integer from an empty string). 
In case of missing json fields, the returned status code should be 400 (bad request).